### PR TITLE
Calculation Vectorization

### PR DESCRIPTION
**Pull request by NoahGallant-MUN**
Vectorized (ie. Improved Efficiency) the Sonci Layer Depth and Sound Channel Axis calculations.

- Benchmarking showed a fairly significant performance improvement.
- Other variables are coming but are proving to be a challenge

### CODE REVIEW

File: data/calculated_parser/functions.py [hunk 1]
```diff
@@ -450,7 +459,47 @@ def criticaldepth(depth, lat, temperature, salinity):
     salinity: The salinity (at all depths) (unitless)
     """
 
+    
     speed = sspeed(depth, lat, temperature, salinity)
+
+
+    """
+    #VECTORIZING -- INCOMPLETE
```
Comment (htmlboss) [inline data/calculated_parser/functions.py]: Large commented-out code blocks should not be put on master @dwayne-hart

File: data/calculated_parser/functions.py [hunk 2]
```diff
@@ -11,6 +11,7 @@
 from pint import UnitRegistry
 from scipy.signal import argrelextrema
 from scipy.stats import linregress
+import time
```
Comment (htmlboss) [inline data/calculated_parser/functions.py]: Is this import being used?

File: data/calculated_parser/functions.py [hunk 3]
```diff
@@ -385,20 +386,29 @@ def soundchannelaxis(depth, lat, temperature, salinity):
...
-            else:
-                result[y,x] = np.nan
-                
-    return result
+    
+    # Mask Array to allow for min index search
+    array = np.ma.masked_array(speed, np.isnan(speed))
+
+    # Find index's of minimum values
+    min_idx = array.argmin(axis = 0)
```
Comment (htmlboss) [inline data/calculated_parser/functions.py]: Could `np.nanargmin` be used here?

https://docs.scipy.org/doc/numpy/reference/generated/numpy.nanargmin.html

File: data/calculated_parser/functions.py [hunk 4]
```diff
@@ -417,27 +427,26 @@ def soniclayerdepth(depth, lat, temperature, salinity):
...
-                        result[y,x] = sld_idx
-                    else:
-                        sld = depth.values[sld_idx]
-                        result[y,x] = sld
-            else:
-                result[y,x] = np.nan
 
-    return result # Only return one horizontal slice
+    marray = np.ma.masked_array(speed, np.isnan(speed))
+    min_idx = marray.argmin(axis=0)
```
Comment (htmlboss) [inline data/calculated_parser/functions.py]: [`np.nanargmin`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.nanargmin.html)

File: data/calculated_parser/functions.py (hunk 1 above)
Comment (htmlboss) [reply data/calculated_parser/functions.py]: If this part won't be completed soon, it should be pulled out of this PR and dumped into another branch w/ draft PR.

File: data/calculated_parser/functions.py (hunk 3 above)
Comment (NoahGallant-MUN) [reply data/calculated_parser/functions.py]: No, this returns a value error when it hits an all NaN slice

File: data/calculated_parser/functions.py (hunk 4 above)
Comment (NoahGallant-MUN) [reply data/calculated_parser/functions.py]: No, all NaN slices (as above)

File: data/calculated_parser/functions.py (hunk 1 above)
Comment (NoahGallant-MUN) [reply data/calculated_parser/functions.py]: Moved code to an issue instead

File: data/calculated_parser/functions.py (hunk 2 above)
Comment (NoahGallant-MUN) [reply data/calculated_parser/functions.py]: No I removed it

**Pull request merged**